bench/bench_htbl.ml

```diff
@@ -0,0 +1,83 @@
+open Multicore_bench
+
+module Key = struct
+  type t = int
+
+  let equal = Int.equal
+  let hash = Fun.id
+end
+
+let run_one ~budgetf ~n_domains ?(n_ops = 20 * Util.iter_factor)
+    ?(n_keys = 10000) ~percent_mem ?(percent_add = (100 - percent_mem + 1) / 2)
+    ?(prepopulate = true) (module Htbl : Htbl_intf.HTBL) =
+  let percent_rem = 100 - (percent_mem + percent_add) in
+
+  let limit_mem = percent_mem in
+  let limit_add = percent_mem + percent_add in
+
+  assert (0 <= limit_mem && limit_mem <= 100);
+  assert (limit_mem <= limit_add && limit_add <= 100);
+
+  let t = Htbl.create ~hashed_type:(module Key) () in
+  if prepopulate then
+    for _ = 1 to n_keys do
+      let value = Random.bits () in
+      let key = value mod n_keys in
+      Htbl.try_add t key value |> ignore
+    done;
```

Comment:
This could be somewhat problematic.  This prepopulates the hash table.  However, the `work` function is run multiple times.  Given that the same random proportion of adds and removes is made the balance should not change much per run, but the balance might still drift over multiple runs.  So, for more consistent results it might make sense to remove everything from the hash table and prepopulate for each run.  (The benchmark in Picos does that.)